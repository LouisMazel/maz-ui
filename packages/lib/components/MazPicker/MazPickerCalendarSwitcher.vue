<template>
  <div class="m-picker-calendar-switcher">
    <MazBtn size="xs" color="transparent" type="button" @click="previousMonth">
      <ChevronLeftIcon class="maz-text-lg" />
    </MazBtn>
    <MazBtn
      size="sm"
      color="transparent"
      type="button"
      class="m-picker-calendar-switcher__date"
      @click="$emit('open-month-switcher', $event)"
    >
      {{ monthLabel }}
    </MazBtn>
    <MazBtn
      size="sm"
      color="transparent"
      type="button"
      class="m-picker-calendar-switcher__date"
      @click="$emit('open-year-switcher', $event)"
    >
      {{ yearLabel }}
    </MazBtn>
    <MazBtn size="xs" color="transparent" type="button" @click="nextMonth">
      <ChevronLeftIcon class="maz-rotate-180 maz-text-lg" />
    </MazBtn>
  </div>
</template>

<script lang="ts" setup>
  import { computed, defineAsyncComponent } from 'vue'
  import { capitalize } from './../../modules/filters/capitalize'
  import { date } from './../../modules/filters/date'
  import dayjs from 'dayjs'

  const MazBtn = defineAsyncComponent(() => import('./../MazBtn.vue'))
  const ChevronLeftIcon = defineAsyncComponent(() => import('./../../icons/chevron-left.svg'))

  const props = defineProps({
    calendarDate: { type: String, default: undefined },
    locale: { type: String, required: true },
    double: { type: Boolean, required: true },
  })

  const emits = defineEmits([
    'previous',
    'next',
    'open-month-switcher',
    'open-year-switcher',
    'update:calendar-date',
  ])

  const calendarDate2 = computed(() => dayjs(props.calendarDate))

  const monthLabel = computed(() => {
    return props.double
      ? `${capitalize(
          date(calendarDate2.value.format(), props.locale, { month: 'long' }),
        )} - ${capitalize(
          date(calendarDate2.value.add(1, 'month').format(), props.locale, {
            month: 'long',
          }),
        )}`
      : capitalize(date(calendarDate2.value.format(), props.locale, { month: 'long' }))
  })

  const yearLabel = computed(() =>
    date(calendarDate2.value.format(), props.locale, { year: 'numeric' }),
  )

  const previousMonth = () => {
    emits('update:calendar-date', dayjs(props.calendarDate).subtract(1, 'month').format())
  }

  const nextMonth = () => {
    emits('update:calendar-date', dayjs(props.calendarDate).add(1, 'month').format())
  }
</script>

<style lang="postcss" scoped>
  .m-picker-calendar-switcher {
    @apply maz-flex maz-space-x-2 maz-border-b maz-border-color-lighter maz-px-2 maz-py-1;

    &__date {
      @apply maz-flex-1 maz-truncate maz-text-center;
    }
  }
</style>
