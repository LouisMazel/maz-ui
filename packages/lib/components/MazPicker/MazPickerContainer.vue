<template>
  <div
    class="m-picker-container"
    :class="{
      '--has-double': double,
      '--is-inline': inline,
      '--has-date': hasDate,
    }"
  >
    <MazPickerHeader
      v-if="!noHeader"
      :color="color"
      :has-time="hasTime"
      :model-value="modelValue"
      :locale="locale"
      :calendar-date="calendarDate"
      :has-date="hasDate"
      :formatter-options="formatterOptions"
      :double="double"
      :no-shortcuts="noShortcuts"
      class="m-picker-container__header"
    />

    <div class="m-picker-container__wrapper">
      <MazPickerCalendar
        v-if="hasDate"
        v-model="currentDate"
        v-model:calendar-date="calendarDate"
        :color="color"
        :locale="locale"
        :has-time="hasTime"
        :double="double"
        :min-date="minDate"
        :max-date="maxDate"
        :first-day-of-week="firstDayOfWeek"
        :disabled="disabled"
        :disabled-weekly="disabledWeekly"
        :disabled-dates="disabledDates"
        :shortcuts="shortcuts"
        :shortcut="shortcut"
        :no-shortcuts="noShortcuts"
        class="m-picker-container__calendar"
      />

      <MazPickerTime
        v-if="hasTime"
        v-model="currentDate"
        v-model:calendar-date="calendarDate"
        :is-open="isOpen"
        :color="color"
        :locale="locale"
        :min-date="minDate"
        :max-date="maxDate"
        :has-date="hasDate"
        :format="format"
        :disabled="disabled"
        :disabled-hours="disabledHours"
        :minute-interval="minuteInterval"
        :formatter-options="formatterOptions"
        :is-hour12="isHour12"
        class="m-picker-container__time"
      />
    </div>
  </div>
</template>

<script lang="ts" setup>
  import { computed, defineAsyncComponent, type PropType } from 'vue'
  import type { Color } from '../types'
  import type { PickerShortcut, PickerValue } from './types'
  import type { DateTimeFormatOptions } from './utils'

  const MazPickerTime = defineAsyncComponent(() => import('./MazPickerTime.vue'))
  const MazPickerHeader = defineAsyncComponent(() => import('./MazPickerHeader.vue'))
  const MazPickerCalendar = defineAsyncComponent(() => import('./MazPickerCalendar.vue'))

  const props = defineProps({
    modelValue: {
      type: [String, Object] as PropType<PickerValue>,
      default: undefined,
    },
    calendarDate: { type: String, required: true },
    color: { type: String as PropType<Color>, required: true },
    locale: { type: String, required: true },
    noHeader: { type: Boolean, default: false },
    firstDayOfWeek: { type: Number, required: true },
    double: { type: Boolean, required: true },
    hasDate: { type: Boolean, required: true },
    minDate: { type: String, default: undefined },
    maxDate: { type: String, default: undefined },
    inline: { type: Boolean, required: true },
    noShortcuts: { type: Boolean, required: true },
    shortcuts: {
      type: Array as PropType<PickerShortcut[]>,
      required: true,
    },
    shortcut: { type: String, default: undefined },
    hasTime: { type: Boolean, required: true },
    isOpen: { type: Boolean, required: true },
    format: { type: String, required: true },
    isHour12: { type: Boolean, required: true },
    formatterOptions: {
      type: Object as PropType<DateTimeFormatOptions>,
      required: true,
    },
    minuteInterval: { type: Number, required: true },
    disabled: { type: Boolean, required: true },
    disabledWeekly: { type: Array as PropType<number[]>, required: true },
    disabledHours: { type: Array as PropType<number[]>, required: true },
    disabledDates: { type: Array as PropType<string[]>, required: true },
  })

  const emits = defineEmits(['update:model-value', 'update:calendar-date', 'close'])

  const currentDate = computed({
    get: () => props.modelValue,
    set: (value) => {
      emits('update:model-value', value)
    },
  })

  const calendarDate = computed({
    get: () => props.calendarDate,
    set: (calendarDate) => emits('update:calendar-date', calendarDate),
  })
</script>

<style lang="postcss" scoped>
  .m-picker-container {
    @apply maz-overflow-hidden maz-rounded maz-bg-color dark:maz-border dark:maz-border-color-light;

    &:not(.--is-inline) {
      @apply maz-absolute maz-z-default-backdrop maz-elevation;
    }

    &.--is-inline {
      @apply maz-border maz-border-color-lighter;
    }

    &.--has-date {
      min-width: 16.875rem;
    }

    &.--has-double {
      min-width: 28.125rem;
    }

    &__wrapper {
      @apply maz-flex;
    }

    & :deep(button):is(:disabled) {
      @apply maz-bg-transparent !maz-text-gray-300 !important;
    }
  }

  html.dark {
    & .m-picker-container {
      & :deep(button):is(:disabled) {
        @apply maz-text-gray-700 !important;
      }
    }
  }
</style>
