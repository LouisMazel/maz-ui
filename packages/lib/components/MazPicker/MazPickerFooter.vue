<template>
  <div class="maz-picker-footer">
    <MazBtn size="mini" :color="color" type="button" @click.stop="$emit('close')">
      <CheckIcon class="icon maz-h-[1.4rem] maz-w-[1.4rem]" />
    </MazBtn>
  </div>
</template>

<script lang="ts" setup>
  import type { PropType } from 'vue'
  import MazBtn from '../MazBtn.vue'
  import type { Color } from '../types'
  import CheckIcon from './../../modules/icons/check.svg'

  defineProps({
    color: { type: String as PropType<Color>, required: true },
  })

  defineEmits(['close'])
</script>

<style lang="postcss" scoped>
  .maz-picker-footer {
    @apply maz-flex maz-justify-end maz-space-x-1 maz-border-t maz-border-color-lighter maz-bg-color maz-p-2;

    & .icon {
      @apply maz-mx-2;
    }
  }

  html.dark .maz-picker-footer {
    @apply maz-bg-color-light;
  }
</style>
