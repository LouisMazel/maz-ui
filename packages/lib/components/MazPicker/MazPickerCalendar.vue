<template>
  <div class="maz-picker-calendar flex">
    <MazPickerShortcuts
      v-if="hasShortcuts"
      v-model="currentValue"
      :color="color"
      :shortcuts="shortcuts"
      :shortcut="shortcut"
      :double="double"
      :disabled="disabled"
    />

    <div class="maz-picker-calendar__main" :class="{ '--has-double': double }">
      <MazPickerCalendarSwitcher
        v-model:calendar-date="calendarDate"
        :locale="locale"
        :double="double"
        @open-month-switcher="monthSwitcherOpen = true"
        @open-year-switcher="yearSwitcherOpen = true"
      />
      <Transition name="maz-picker-slide">
        <MazPickerMonthSwitcher
          v-if="monthSwitcherOpen"
          v-model:calendar-date="calendarDate"
          :color="color"
          :double="double"
          :locale="locale"
          @close="monthSwitcherOpen = false"
        />
      </Transition>
      <Transition name="maz-picker-slide">
        <MazPickerYearSwitcher
          v-if="yearSwitcherOpen"
          v-model:calendar-date="calendarDate"
          :color="color"
          :locale="locale"
          @close="yearSwitcherOpen = false"
        />
      </Transition>
      <div class="maz-picker-calendar__months">
        <MazPickerCalendarMonth
          v-for="month in months"
          :key="month"
          v-model="currentValue"
          v-model:hoverred-day="hoverredDay"
          :calendar-date="calendarDate"
          :locale="locale"
          :has-time="hasTime"
          :color="color"
          :offset-month="month"
          :first-day-of-week="firstDayOfWeek"
          :min-date="minDate"
          :max-date="maxDate"
          :disabled="disabled"
          :disabled-weekly="disabledWeekly"
          :disabled-dates="disabledDates"
        />
      </div>
    </div>
  </div>
</template>

<script lang="ts" setup>
  import { computed, type PropType, ref, defineAsyncComponent } from 'vue'
  import type { Color } from '../types'
  import type { PickerShortcut, PickerValue } from './types'
  import type { Dayjs } from 'dayjs'

  import MazPickerCalendarSwitcher from './MazPickerCalendarSwitcher.vue'
  import MazPickerCalendarMonth from './MazPickerCalendarMonth/MazPickerCalendarMonth.vue'

  const MazPickerMonthSwitcher = defineAsyncComponent(() => import('./MazPickerMonthSwitcher.vue'))
  const MazPickerYearSwitcher = defineAsyncComponent(() => import('./MazPickerYearSwitcher.vue'))
  const MazPickerShortcuts = defineAsyncComponent(() => import('./MazPickerShortcuts.vue'))

  const props = defineProps({
    modelValue: {
      type: [String, Object] as PropType<PickerValue>,
      default: undefined,
    },
    calendarDate: { type: String, required: true },
    color: { type: String as PropType<Color>, required: true },
    locale: { type: String, required: true },
    firstDayOfWeek: { type: Number, required: true },
    double: { type: Boolean, required: true },
    minDate: { type: String, default: undefined },
    maxDate: { type: String, default: undefined },
    disabledWeekly: { type: Array as PropType<number[]>, required: true },
    disabledDates: { type: Array as PropType<string[]>, required: true },
    shortcuts: {
      type: Array as PropType<PickerShortcut[]>,
      required: true,
    },
    noShortcuts: { type: Boolean, required: true },
    hasTime: { type: Boolean, required: true },
    shortcut: { type: String, default: undefined },
    disabled: { type: Boolean, required: true },
  })

  const emits = defineEmits(['update:model-value', 'update:calendar-date'])

  const hoverredDay = ref<Dayjs>()

  const isRangeMode = computed(() => props.modelValue && typeof props.modelValue === 'object')

  const hasShortcuts = computed(
    () => !props.noShortcuts && props.shortcuts.length > 0 && isRangeMode.value,
  )

  const monthSwitcherOpen = ref(false)
  const yearSwitcherOpen = ref(false)

  const currentValue = computed({
    get: () => props.modelValue,
    set: (value) => emits('update:model-value', value),
  })

  const months = computed(() => Array.from({ length: props.double ? 2 : 1 }, (_v, i) => i))

  const calendarDate = computed({
    get: () => props.calendarDate,
    set: (calendarDate) => emits('update:calendar-date', calendarDate),
  })
</script>

<style lang="postcss" scoped>
  .maz-picker-calendar {
    @apply maz-relative maz-flex maz-w-full;

    &__main {
      @apply maz-flex maz-flex-1 maz-flex-col;

      width: 16rem;

      &.--has-double {
        width: 30rem;

        & .maz-picker-calendar__months > *:first-child {
          @apply maz-border-r maz-border-color-lighter;
        }
      }
    }

    &__months {
      @apply maz-flex maz-w-full maz-flex-1;
    }
  }

  .maz-picker-slide-enter-active,
  .maz-picker-slide-leave-active {
    transition: all 300ms ease-in-out;
    transform: translateY(0);
  }

  .maz-picker-slide-enter-from,
  .maz-picker-slide-leave-to {
    transform: translateY(100%);
  }
</style>
