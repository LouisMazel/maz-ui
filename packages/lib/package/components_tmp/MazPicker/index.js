import MazPicker from './_main.vue'

MazPicker.install = (Vue) => {
  Vue.component(MazPicker.name, MazPicker)
}

export default MazPicker
