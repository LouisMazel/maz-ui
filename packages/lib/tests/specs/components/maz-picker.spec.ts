import { mount } from '@vue/test-utils'
import MazPicker from '@components/MazPicker.vue'

test('components/MazPicker.vue', () => {
  expect(MazPicker).toBeTruthy()

  const wrapper = mount(MazPicker, {})

  expect(wrapper.html()).toMatchSnapshot()
})
