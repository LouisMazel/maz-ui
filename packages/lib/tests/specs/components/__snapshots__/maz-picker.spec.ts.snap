// Vitest Snapshot v1

exports[`components/MazPicker.vue 1`] = `
"<div class=\\"m-picker m-picker--primary m-picker--bottom m-picker--left\\" data-v-43f83472=\\"\\">
  <div class=\\"m-input --is-readonly m-picker__input --primary --md\\" data-v-62586b96=\\"\\" data-v-43f83472=\\"\\">
    <div class=\\"m-input-wrapper maz-border --default-border maz-rounded-lg\\" data-v-62586b96=\\"\\">
      <!--v-if-->
      <div class=\\"m-input-wrapper-input\\" data-v-62586b96=\\"\\"><input id=\\"MazInput\\" type=\\"text\\" name=\\"input\\" autocomplete=\\"off\\" class=\\"m-picker__input m-input-input\\" readonly=\\"\\" data-v-62586b96=\\"\\">
        <!--v-if-->
      </div>
      <div class=\\"m-input-wrapper-right\\" data-v-62586b96=\\"\\"><button tabindex=\\"-1\\" class=\\"m-picker__button\\" data-v-43f83472=\\"\\">
          <!-- eslint-disable vue/no-v-html -->
          <!--v-if-->
          <!-- eslint-enable vue/no-v-html -->
        </button>
        <!--v-if-->
        <!--v-if-->
      </div>
    </div>
  </div>
  <transition-stub data-v-43f83472=\\"\\">
    <div class=\\"m-picker-container --has-date\\" id=\\"mazPickerContainer-1\\" data-v-5d9b7c5a=\\"\\" data-v-43f83472=\\"\\" style=\\"display: none;\\">
      <div class=\\"m-picker-header --primary m-picker-container__header\\" data-v-bf7ebef8=\\"\\" data-v-5d9b7c5a=\\"\\">
        <div class=\\"m-picker-header__date\\" data-v-bf7ebef8=\\"\\">
          <transition-group-stub data-v-bf7ebef8=\\"\\"><span class=\\"m-picker-header__year-text\\" data-v-bf7ebef8=\\"\\">-</span></transition-group-stub>
          <transition-group-stub data-v-bf7ebef8=\\"\\"><span class=\\"m-picker-header__date-text\\" data-v-bf7ebef8=\\"\\">-</span></transition-group-stub>
        </div>
        <!--v-if-->
      </div>
      <div class=\\"m-picker-container__wrapper\\" data-v-5d9b7c5a=\\"\\">
        <div class=\\"maz-picker-calendar flex m-picker-container__calendar\\" data-v-44fdfcf5=\\"\\" data-v-5d9b7c5a=\\"\\">
          <!--v-if-->
          <div class=\\"maz-picker-calendar__main\\" data-v-44fdfcf5=\\"\\">
            <div class=\\"m-picker-calendar-switcher\\" data-v-f593db14=\\"\\" data-v-44fdfcf5=\\"\\"><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-f593db14=\\"\\">
                <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><!-- eslint-disable vue/no-v-html --><!--v-if--><!-- eslint-enable vue/no-v-html --></span>
                <!--v-if-->
                <!--v-if-->
              </button><button class=\\"m-picker-calendar-switcher__date m-btn --sm --transparent --cursor-pointer --is-button m-picker-calendar-switcher__date\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-f593db14=\\"\\">
                <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\">May</span>
                <!--v-if-->
                <!--v-if-->
              </button><button class=\\"m-picker-calendar-switcher__date m-btn --sm --transparent --cursor-pointer --is-button m-picker-calendar-switcher__date\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-f593db14=\\"\\">
                <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\">2022</span>
                <!--v-if-->
                <!--v-if-->
              </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-f593db14=\\"\\">
                <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><!-- eslint-disable vue/no-v-html --><!--v-if--><!-- eslint-enable vue/no-v-html --></span>
                <!--v-if-->
                <!--v-if-->
              </button></div>
            <transition-stub data-v-44fdfcf5=\\"\\">
              <!--v-if-->
            </transition-stub>
            <transition-stub data-v-44fdfcf5=\\"\\">
              <!--v-if-->
            </transition-stub>
            <div class=\\"maz-picker-calendar__months\\" data-v-44fdfcf5=\\"\\">
              <div class=\\"maz-picker-calendar-month\\" data-v-42c61b5c=\\"\\" data-v-44fdfcf5=\\"\\">
                <div class=\\"maz-picker-calendar-days maz-picker-calendar-month__days\\" data-v-a15a9a16=\\"\\" data-v-42c61b5c=\\"\\"><span data-v-a15a9a16=\\"\\">Sun</span><span data-v-a15a9a16=\\"\\">Mon</span><span data-v-a15a9a16=\\"\\">Tue</span><span data-v-a15a9a16=\\"\\">Wed</span><span data-v-a15a9a16=\\"\\">Thu</span><span data-v-a15a9a16=\\"\\">Fri</span><span data-v-a15a9a16=\\"\\">Sat</span></div>
                <div class=\\"maz-picker-calendar-grid\\" data-v-4e1f3e64=\\"\\" data-v-42c61b5c=\\"\\">
                  <transition-group-stub data-v-4e1f3e64=\\"\\">
                    <div class=\\"maz-picker-calendar-grid__container\\" data-v-4e1f3e64=\\"\\"><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">1</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">2</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">3</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">4</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">5</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">6</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">7</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">8</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">9</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">10</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">11</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">12</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">13</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">14</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">15</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">16</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"--is-today m-btn --mini --transparent --cursor-pointer --is-button --is-today\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">17</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">18</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">19</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">20</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">21</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">22</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">23</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">24</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">25</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">26</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">27</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">28</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">29</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">30</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button><button class=\\"m-btn --mini --transparent --cursor-pointer --is-button\\" type=\\"button\\" data-v-486cf628=\\"\\" data-v-4e1f3e64=\\"\\">
                        <!--v-if--><span class=\\"maz-flex maz-flex-center\\" data-v-486cf628=\\"\\"><span data-v-4e1f3e64=\\"\\">31</span></span>
                        <!--v-if-->
                        <!--v-if-->
                      </button></div>
                  </transition-group-stub>
                </div>
              </div>
            </div>
          </div>
        </div>
        <!--v-if-->
      </div>
    </div>
  </transition-stub>
</div>"
`;
